engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/typer/TypeMethodReference.scala

```diff
@@ -7,22 +7,20 @@ import pl.touk.nussknacker.engine.definition.TypeInfos.{ClazzDefinition, MethodI
 import pl.touk.nussknacker.engine.types.EspTypeUtils
 
 object TypeMethodReference {
-  def apply(methodName: String, currentResults: List[TypingResult], params: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean)(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
-    new TypeMethodReference(methodName, currentResults, params, typeDefinitionSet, disableMethodExecutionForUnknown).call
+  def apply(methodName: String, invocationTarget: TypingResult, params: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean)(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
+    new TypeMethodReference(methodName, invocationTarget, params, typeDefinitionSet, disableMethodExecutionForUnknown).call
 }
 
-class TypeMethodReference(methodName: String, currentResults: List[TypingResult], calledParams: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean) {
+class TypeMethodReference(methodName: String, invocationTarget: TypingResult, calledParams: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean) {
   def call(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
-    currentResults.headOption match {
-      case Some(tc: SingleTypingResult) =>
+    invocationTarget match {
+      case tc: SingleTypingResult =>
         typeFromClazzDefinitions(extractClazzDefinitions(Set(tc)))
-      case Some(TypedUnion(nestedTypes)) =>
+      case TypedUnion(nestedTypes) =>
         typeFromClazzDefinitions(extractClazzDefinitions(nestedTypes))
-      case _ =>
-        if(disableMethodExecutionForUnknown)
-          Left("Method access on Unknown type is blocked")
-        else
-          Right(Unknown)
+      case Unknown =>
+        if(disableMethodExecutionForUnknown) Left("Method invocation on Unknown is not allowed") else Right(Unknown)
+      case _ => Right(Unknown)
```

Comment:
I think this is not needed ATM. There are no more cases - if there are, it should be catched by compiler

Reply:
Done